src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useIsApprovalNeeded.tsx

```diff
@@ -6,24 +6,54 @@ import { useCallback, useMemo } from 'react'
 import { reactQueries } from 'react-queries'
 import { selectAllowanceCryptoBaseUnit } from 'react-queries/hooks/selectors'
 import type { GetAllowanceErr } from 'react-queries/types'
-import { bn } from 'lib/bignumber/bignumber'
+import { usdtAssetId } from 'components/Modals/FiatRamps/config'
+import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import { bn, bnOrZero } from 'lib/bignumber/bignumber'
+import { selectRelatedAssetIdsInclusive } from 'state/slices/related-assets-selectors'
+import { useAppSelector } from 'state/store'
 
 export const useIsApprovalNeeded = (
   tradeQuoteStep: TradeQuoteStep | undefined,
   sellAssetAccountId: AccountId | undefined,
 ) => {
+  const relatedAssetIdsFilter = useMemo(() => ({ assetId: usdtAssetId }), [])
+  const usdtAssetIds = useAppSelector(state =>
+    selectRelatedAssetIdsInclusive(state, relatedAssetIdsFilter),
+  )
+  const isUsdtApprovalResetEnabled = useFeatureFlag('UsdtApprovalReset')
   const selectIsApprovalNeeded = useCallback(
     (data: Result<string, GetAllowanceErr>) => {
       if (tradeQuoteStep === undefined) return undefined
 
       const allowanceCryptoBaseUnit = selectAllowanceCryptoBaseUnit(data)
-      return allowanceCryptoBaseUnit !== undefined
-        ? bn(allowanceCryptoBaseUnit).lt(
-            tradeQuoteStep.sellAmountIncludingProtocolFeesCryptoBaseUnit,
-          )
-        : false
+      const isApprovalNeeded =
+        allowanceCryptoBaseUnit !== undefined
+          ? bn(allowanceCryptoBaseUnit).lt(
+              tradeQuoteStep.sellAmountIncludingProtocolFeesCryptoBaseUnit,
+            )
+          : false
```

Comment:
nitpick - fallback to empty string instead of ternary to make it more readable imo
```suggestion
      const isApprovalNeeded = bn(allowanceCryptoBaseUnit ?? '').lt(
        tradeQuoteStep.sellAmountIncludingProtocolFeesCryptoBaseUnit,
      )
```

Reply:
Ah true `bn('')` is `NaN`, so it will always implicitly evaluate to `true` as `NaN < number` is always false.

Not super clear logically but code-wise cleaner so I'll do that and add a code comment for clarity.

Reply:
Actually I think relying on empty string internally evaluating to NaN in bignumber.js is less clear than explicit ternary here, so will leave this one as-is